tests/ha/single_pod_restart/test_server_pod_restart.py

```diff
@@ -0,0 +1,398 @@
+        resp = self.ha_obj.delete_kpod_with_shutdown_methods(
+            master_node_obj=self.node_master_list[0], health_obj=self.hlth_master_list[0],
+            down_method=const.RESTORE_DEPLOYMENT_K8S, pod_prefix=[const.SERVER_POD_NAME_PREFIX])
+        assert_utils.assert_true(resp[1], "Failed to shutdown/delete server pod")
+        pod_name = list(resp[1].keys())[0]
+        self.deployment_name = resp[1][pod_name]['deployment_name']
+        self.deployment_backup = resp[1][pod_name]['deployment_backup']
+        self.restore_method = resp[1][pod_name]['method']
+        assert_utils.assert_true(resp[0], "Cluster/Services status is not as expected")
+        LOGGER.info("Step 2: Successfully shutdown server pod %s. Verified cluster and "
+                    "services states are as expected & remaining pods status is online.", pod_name)
+        self.restore_pod = True
+        LOGGER.info("Step 3: Perform WRITEs with variable object sizes")
+        self.test_prefix_deg = 'test-44836-deg'
+        resp = self.ha_obj.ha_s3_workload_operation(s3userinfo=list(users.values())[0],
+                                                    log_prefix=self.test_prefix_deg,
+                                                    skipread=True, skipcleanup=True,
+                                                    setup_s3bench=False)
+        assert_utils.assert_true(resp[0], resp[1])
+        LOGGER.info("Step 3: Performed WRITEs with variable sizes objects.")
+        LOGGER.info("Step 4: Restore server pod and check cluster status.")
+        resp = self.ha_obj.restore_pod(pod_obj=self.node_master_list[0],
+                                       restore_method=self.restore_method,
+                                       restore_params={"deployment_name": self.deployment_name,
+                                                       "deployment_backup":
+                                                           self.deployment_backup},
+                                       clstr_status=True)
+        LOGGER.debug("Response: %s", resp)
+        assert_utils.assert_true(resp[0], f"Failed to restore server pod by {self.restore_method} "
+                                          "way OR the cluster is not online")
+        LOGGER.info("Step 4: Successfully started server pod and cluster is online.")
+        self.restore_pod = False
+        LOGGER.info("Step 5: Perform READs and verify DI on the written data with buckets created "
+                    "in healthy cluster.")
+        resp = self.ha_obj.ha_s3_workload_operation(s3userinfo=list(users.values())[0],
+                                                    log_prefix=self.test_prefix, skipwrite=True,
+                                                    skipcleanup=True, setup_s3bench=False)
+        assert_utils.assert_true(resp[0], resp[1])
+        LOGGER.info("Step 5: Successfully run READ/Verify on data written with buckets created "
+                    "in healthy cluster")
```

Comment:
I think instead of doing read-verify on data written in healthy cluster, we should try to write data on buckets created in healthy cluster, as this is write related test case.

Reply:
Agree

Reply:
And same goes for buckets created in degraded cluster